### ISSUE TITLE
Early stopping on RL problem rollouts

### ISSUE THREAD

**Issue by keraJLi** [RobertTLange/evosax#97]
RL problem rollouts use `jax.lax.scan` to roll out for a full `self.episode_length` steps, flagging steps as invalid once a vmapped episode has finished (see e.g. here in [GymnaxProblem](https://github.com/RobertTLange/evosax/blob/a213f27485803f13e70fb2e4f3c9756176ca4c71/evosax/problems/rl/gymnax.py#L141)). This can be problematic if episodes have a big time limit. For example, craftax uses 100000 by default, but episodes typically end after a few hundred steps.

We can implement early stopping by using `jax.lax.while`, as rejax does in its [evaluate.py](https://github.com/keraJLi/rejax/blob/f1afadd1fc7e60dfe3145000df35ebf64a970a17/src/rejax/evaluate.py). I am happy to create a pull request, once I have the time.

**Comment by maxencefaldor** [comment]
Thanks for the suggestion. You're right that `jax.lax.while` is more efficient for rollouts than `scan`.

However, we are intentionally keeping the `Problem` classes in `evosax` minimal. The library's focus is on maintaining the **optimizers** (e.g., CMA-ES, OpenAI-ES, etc.), not on building feature-rich **problems**. The bundled problems are just simple examples.

We wouldn't be merging this feature to avoid scope creep.

The intended way to handle this is to create your own `Problem` class (or just a simple function) that implements your desired logic (like `while_loop`). The `evosax` `ask`/`tell` interface is decoupled from the problem, so you are free to generate the fitness scores however you see fit.

**Comment by keraJLi** [comment]
Hi @maxencefaldor. This is not a new feature, but a different way to implement observation normalization. 

I have made the necessary changes, **without increasing implementation complexity**:
* `gymnax.py`: 63 insertions(+), 61 deletions(-)
* `brax.py`: 68 insertions(+), 64 deletions(-)
So +6 lines total, even though my code is very verbose.

I have tested both.

Caveat: we don't return all env states of the complete rollout. Doing this might be nice for visualization, but then we would also have to return state validity and mask out invalid states, which is not currently implemented and might easily be forgotten by users of the library. I suggest rolling out policies again for visualization, which is orders of magnitude cheaper than running a single generation.
